File: BugTracker.API/Models/Task.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace BugTracker.API.Models
{
    public class Task
    {
        public int Id { get; set; }
        public Project Project { get; set; }
        [Required]
        public int ProjectId { get; set; }
        [Required]
        public DateTime StartDate { get; set; }
        public DateTime DueDate { get; set; }
        public Employee Worker { get; set; }
        [Required]
        public int WorkerId { get; set; }
        public int GiverId { get; set; }
        [Required]
        public int PercentageDone { get; set; }
        public string Description { get; set; }
        [Required]
        public string Title { get; set; }
        public TaskStatus Status { get; set; }

    }
}

File: BugTracker-SPA/src/app/model/task.ts
import { Project } from './project';
import { Employee } from './employee';
export class Task {
    id: number;
    projectId: number;
    project: Project;
    startDate: Date;
    dueDate: Date;
    workerId: number;
    worker: Employee;
    giverId: number;
    percentageDone: number;
    description: string;
    title: string;
}
